### DISCUSSION TITLE
t-watch-2020 V3 LCD problem

### DISCUSSION THREAD

**Discussion by wovo** [micropython/micropython#9547] [Hardware & Peripherals]
I am tinkering with a t-watch-2020, specifically a version 3. I can use the motor, backlight, touch, gravity, and RTC, but NOT the LCD, not even with code that worked with an older version (V1 IIRC). The manufacturer demo does work, so the LCD is not broken. As far as I can see the LCD connections are exactly the same for V1 and V3. (I know the backlight pin is different.)

Is there some difference I missed?

Has anyone seen some known-working code for a V3?

**Comment by andypiper** [comment]
What MicroPython build are you using?

I have something simple working but I’ll need to grab it off another machine. Pretty sure mine is a v3 as well. Making a mental not to dig it out in the next couple of days.

**Reply by wovo** [reply]
I use esp32spiram-2022618-v1.19.1

V3 is the one with microphone (not GPS, that is V2), and backlight at pin 15 (V1: 12, V2: 25). (Why do they keep changing that pin?)

**Comment by picpic020960** [comment]
Bonjour ,
with t-watch-2020-v2 (GPS and SDCARD) I have issues with st7789 (TFT) and  touch. test with I2C and SPI 'Soft-' or not , KO.
I have test the [https://github.com/russhughes/st7789s3_mpy](url) , some results but bad color.
I2C.scan answer : nothing (KO ==> reset) with HW I2C and  [] (empty list ) with SoftI2C four touch.
for PCF , BMA, APX , DRV(aptic) I2C.scan OK :
`>>> from machine import I2C , Pin
>>> i2c = I2C(1, scl=Pin(22), sda=Pin(21))
>>> i2c.scan()
[25, 53, 81, 90]`

if you have some ideas or tricks , thanks
